### PR TITLE
 Use libp2p's implementation of a wasm websocket transport 

### PR DESCRIPTION
**Pull request by expenses**
Builds on top of #5088.

### CODE REVIEW

**Comment by tomaka** [comment]
So... that's technically a breaking change.

**Comment by expenses** [comment]
@tomaka oh yeah, I'll make a polkadot PR too.

File: utils/browser/src/lib.rs [hunk 1]
```diff
@@ -26,25 +26,23 @@ use futures::{prelude::*, channel::{oneshot, mpsc}, future::{poll_fn, ok}, compa
...
-/// Create a service configuration from a chain spec and the websocket transport.
+/// Create a service configuration from a chain spec.
 ///
 /// This configuration contains good defaults for a browser light client.
-pub async fn browser_configuration<G, E>(
-	transport: Transport,
-	chain_spec: ChainSpec<G, E>,
-) -> Result<Configuration<G, E>, Box<dyn std::error::Error>>
+pub async fn browser_configuration<G, E>(chain_spec: ChainSpec<G, E>)
+-> Result<Configuration<G, E>, Box<dyn std::error::Error>>
```
Comment (NikVolf) [inline utils/browser/src/lib.rs]:
```suggestion
	-> Result<Configuration<G, E>, Box<dyn std::error::Error>>
```

File: bin/node/cli/src/browser.rs [hunk 2]
```diff
@@ -19,26 +19,26 @@ use log::info;
...
 
-async fn start_inner(chain_spec: String, log_level: String, wasm_ext: Transport) -> Result<Client, Box<dyn std::error::Error>> {
+async fn start_inner(chain_spec: String, log_level: String) -> Result<Client, Box<dyn std::error::Error>> {
 	set_console_error_panic_hook();
 	init_console_log(log::Level::from_str(&log_level)?)?;
 	let chain_spec = ChainSpec::from_json_bytes(chain_spec.as_bytes().to_vec())
 		.map_err(|e| format!("{:?}", e))?;
 
-	let config: Configuration<_, _> = browser_configuration(wasm_ext, chain_spec)
+	let config: Configuration<_, _> = browser_configuration(chain_spec)
```
Comment (NikVolf) [inline bin/node/cli/src/browser.rs]: is `: Configuration<_, _>` annotation still required ?

if you tell me how to compile it locally, I'll check myself :)

Comment (expenses) [reply bin/node/cli/src/browser.rs]: `cargo check --target wasm32-unknown-unknown -p node-cli --no-default-features --features browser` should do it.

Comment (expenses) [reply bin/node/cli/src/browser.rs]: Done

**Comment by expenses** [comment]
Hey, I'd like to get this reviewed and merged soon if possible.

File: utils/browser/src/lib.rs [hunk 3]
```diff
@@ -26,25 +26,23 @@ use futures::{prelude::*, channel::{oneshot, mpsc}, future::{poll_fn, ok}, compa
 use std::task::Poll;
 use std::pin::Pin;
 use sc_chain_spec::Extension;
+use libp2p_wasm_ext::{ExtTransport, ffi};
```
Comment (tomaka) [inline utils/browser/src/lib.rs]: (alphabetical ordering?)

**Comment by gnunicorn** [comment]
could you add the appropriate b-label?

**Pull request merged**